master/internal/core.go

```diff
@@ -976,6 +977,21 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 		return errors.Wrap(err, "could not fetch cluster id from database")
 	}
 
+	if isBrandNewCluster {
+		if password := m.config.Security.InitialUserPassword; password == "" {
+			log.Warn("This cluster was deployed without a default password for the built-in `determined` " +
+				"and `admin` users. You should set one using `det user change-password`. New clusters can be " +
+				"deployed with default passwords set using the DET_SECURITY_INITIAL_USER_PASSWORD setting.")
```

Comment:
Should it not be `INITIAL_USER_PASSWORD`? That's what we called it earlier in the tests. Could we also specify here that it can be set with the cli arg: `--initial-user-password`.

Also, the user can't use `det user change-password` if they weren't already admin correct? I don't know if telling them they can use that path makes sense.

Reply:
> Should it not be INITIAL_USER_PASSWORD? That's what we called it earlier in the tests.

This is master code not det deploy, so the format changes (it is prefixed to avoid collisions). I didn't prefix it elsewhere because we know what is in CI and the CI var is just fed to `--initial-user-password`.

> Could we also specify here that it can be set with the cli arg: --initial-user-password.

This flag only applies to det deploy. The master flag would `--security-initial-user-password` I think. Maybe I'll just replace the env var with the config name, and folks that know they can set with flags or envvars in special ways will just know.

> Also, the user can't use det user change-password if they weren't already admin correct? I don't know if telling them they can use that path makes sense.

A password can be changed by the user for that password or the admin. Also this is in cluster logs so it really is just for admins anyway.